src/kaleidoscope/key_defs.h

```diff
@@ -255,14 +241,15 @@ typedef kaleidoscope::Key Key_;
 #define KEY_LEFT_FN2 0xff
 #define Key_LFN2 Key(KEY_LEFT_FN2, KEY_FLAGS)
 
+#define SYSTEM_KEY(code) Key(code, SYNTHETIC | IS_SYSCTL)
 
 /* Most Consumer keys are more then 8bit, the highest Consumer hid code
    uses 10bit. By using the 11bit as flag to indicate a consumer keys was activate we can
    use the 10 lsb as the HID Consumer code. If you need to get the keycode of a Consumer key
    use the CONSUMER(key) macro this will return the 10bit keycode.
 */
 #define CONSUMER(key) (key.getRaw() & 0x03FF)
-#define CONSUMER_KEY(code, flags) Key((code) | ((flags | SYNTHETIC|IS_CONSUMER) << 8))
+#define CONSUMER_KEY(code) Key((code & 0x03FF) | ((SYNTHETIC | IS_CONSUMER) << 8))
```

Comment:
This is a simpler and more correct version of `CONSUMER_KEY()`. The high six bits should always be the same on any Consumer key, regardless of HID usage type.

Really, I think this should be a `static constexpr` function inside the `Key` class: `Key::consumer(uint16_t code)`, rather than a macro, but that requires more extensive code reorganization.

Reply:
Now that we're using that magic 0x03FF constant more than once, It should get defined as a named constant. Really it should have been one from the beginning.

Reply:
Excellent point. I'll get on that.